apis/kueue/validation/validation.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// This is mostly inspired by https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/v1/validation/validation.go#ValidateContainerResourceName
```

Comment:
this validation feels like something we can move to `k8s.io/apimachinery/pkg/util/validation` as followup.

Reply:
it is actually scary and likely will diverge from the one in k8s at some point. May be we shouldn't validate the resource name at all and avoid this complexity.

Reply:
since we only match the pod against this field (vs creating pods out of it), I think we should be ok if we are not strict. It is probably enough if we just do validate that the name `IsQualifiedName`, a function already exported by k8s

Reply:
> this validation feels like something we can move to `k8s.io/apimachinery/pkg/util/validation` as followup.

Yes, I was going to open an issue to discuss about this.

Reply:
I talked offline with liggit@, this is too complicated to move as is. We don't necessarily need strict validation here, please just simplify it only validate that the name is qualified

Reply:
Updated as advised.